src/main/java/net/minecraftforge/event/world/BlockEvent.java

```diff
@@ -161,6 +162,23 @@ public void setExpToDrop(int exp)
     }
 
     /**
+     * Called when a block is destroyed by an explosion.
+     * Canceling this event will prevent the block from being destroyed.
+     */
+    @Cancelable
+    public static class ExplodeEvent extends BlockEvent
+    {
+        @Nonnull private final Explosion explosion;
```

Comment:
1. This formatting is wrong
2. Package is already non-null, so you can remove this annotation